File: backend/src/main/java/com/back/domain/member/member/service/AuthTokenService.java
package com.back.domain.member.member.service;

import com.back.domain.member.member.entity.Member;
import com.back.global.standard.util.Ut;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;


import java.util.Map;

@Service
public class AuthTokenService {
    @Value("${custom.jwt.secretKey}")
    private String jwtSecretKey;

    @Value("${custom.accessToken.expirationSeconds}")
    private int accessTokenExpSec;

    @Value("${custom.refreshToken.expirationSeconds}")
    private int refreshTokenExpSec;

    public String genAccessToken(Member member) {
        int id = member.getId();
        String email = member.getEmail();

        return Ut.jwt.toString(
                jwtSecretKey,
                accessTokenExpSec,
                Map.of(
                        "id",id,
                        "email",email
                )
        );
    }

    public String genRefreshToken(Member member) {
        int id = member.getId();
        String email = member.getEmail();

        return Ut.jwt.toString(
                jwtSecretKey,
                refreshTokenExpSec,
                Map.of("id", id,
                        "email",email));
    }

    
    public Map<String, Object> payload(String accessToken) {
        Map<String, Object> parsedPayload = Ut.jwt.payload(jwtSecretKey, accessToken);

        if(parsedPayload == null) return null;
        int id = (int) parsedPayload.get("id");
        String email = (String) parsedPayload.get("email");

        return Map.of("id", id, "email", email);
    }
    
    public boolean isValid(String accessToken) { 
        return Ut.jwt.isValid(jwtSecretKey, accessToken); 
    }
}

File: backend/src/main/java/com/back/domain/member/member/service/AuthTokenService.kt
package com.back.domain.member.member.service

import com.back.domain.member.member.entity.Member
import com.back.global.standard.util.Ut
import org.springframework.beans.factory.annotation.Value

import org.springframework.stereotype.Service
import java.util.Map

@Service
class AuthTokenService(
    @Value("\${custom.jwt.secretKey}")
    private val jwtSecretKey: String,
    @Value("\${custom.accessToken.expirationSeconds}")
    private val accessTokenExpSec: Int,
    @Value("\${custom.refreshToken.expirationSeconds}")
    private val refreshTokenExpSec: Int
) {
   fun genAccessToken(member: Member): String? {
        val payload = mapOf(
            "id" to member.id,
            "email" to member.getEmail()
        )

        return Ut.jwt.toString(
            jwtSecretKey,
            accessTokenExpSec,
            payload
        )
    }

    fun genRefreshToken(member: Member): String? {
        val payload = mapOf(
            "id" to member.id,
            "email" to member.getEmail()
        )

        return Ut.jwt.toString(
            jwtSecretKey,
            refreshTokenExpSec,
            payload
        )
    }


    fun payload(accessToken: String): MutableMap<String, Any>? {
        val parsedPayload = Ut.jwt.payload(jwtSecretKey, accessToken)
            ?: return null

        return mutableMapOf(
            "id" to parsedPayload["id"] as Int,
            "email" to parsedPayload["email"] as String
        )
    }

    fun isValid(accessToken: String): Boolean {
        return Ut.jwt.isValid(jwtSecretKey, accessToken)
    }
}
